### DISCUSSION TITLE
Geocache organization ideas

### DISCUSSION THREAD

**Discussion by elreydetoda** [JupiterBroadcasting/network#13] [Ideas]
So, geocaches have been floating around the network for the past couple of weeks and talk of having a way for JB to organize that. While I love the hand-crafted, in-person updates that happened (https://podverse.fm/clip/8NdMyaRuS) :sweat_smile:, I figure a way of automatically including this in some website would make it easier to track.  I figured creating this issue could help consolidate ideas in one location and allow people to vote for the ideas they like best.

Based on the latter part of the clip above, identifying a way for listeners to submit requests for JB swag so they can put a geocache near them would be a requirement as well.

**Comment by wwff** [comment]
您好！您的邮件我已收到

**Comment by cbeauhilton** [comment]
One solution that is pretty easy to implement, doesn’t require  any auth providers (self-hosted or otherwise): 

https://github.com/Greenheart/pagecrypt

This is password protection for html pages that is easy to incorporate into static sites. You provide a password/phrase/etc, the page is encrypted with that key, then when you visit the page you get a little dialogue box. “Inspect page” just gives you the dialogue box and a bunch of encrypted gobbledegook.

I usually specify a header in the .md, e.g. ‘password_protect: true’, then grep for it and run those pages through pagecrypt after build but prior to publication. Provide the passphrase as an environment variable, easy to switch out after a cache has been found and you want to rotate keys.

Would have to have that repository private, probably use a subdomain, or use a GitHub Action on this site to pull the md from elsewhere, so the beans aren’t spilt by looking at the source code of this repo.

One workflow would be to have a QR code with a long passphrase per geocache, which takes you to a special page with further instructions and some Easter eggs, maybe a printable QR code with the next planned key for that cache. 

You can also include the passphrase as part of the URL params, so you get shareable URLs that don’t require the user to actually type anything in.

Or you could skip all of this and make it so the QR code just takes you to a regular contact page with a prefilled token (something the mail filters could keep an eye out for), which you submit and then get an email from JB asking if you have any vids/pics to share, via whatever barbaric method is most preferable for the user.

Of course, a little FastAPI/htmx/etc action is always great. :)

**Comment by elreydetoda** [comment]
Convo in matrix ([src](https://matrix.to/#/!rDlEtonuSythLzmxQL:jupiterbroadcasting.com/$hDhKwhggXvcp39ehWN71Polo82adkCKGODnoDd3jUt4)):
> On geocaching, I have a personal project using leaflet and OSM the initial idea was to rewrite wikipedias geohack service and include a map that was clickable. Code is not yet public but you can take a look at
https://geo.johnstonetech.co.uk/ i think it would be nice to have geocaches displayed on a map like leaflet or openlayers. So if i [can.be](http://can.be/) any help with that let me know.
